hail/src/main/scala/is/hail/io/fs/AzureStorageFS.scala

```diff
@@ -0,0 +1,280 @@
+      }
+
+      override def seek(newPos: Long): Unit = {
+        bb.clear()
+        bb.limit(0)
+        pos = newPos
+      }
+    }
+
+    new WrappedSeekableDataInputStream(is)
+  }
+
+  def createNoCompression(filename: String): PositionedDataOutputStream = {
+    val (account, container, path) = getAccountContainerPath(filename)
+    val appendClient: AppendBlobClient = getBlobClient(account, container, path).getAppendBlobClient
+    if (!appendClient.exists()) appendClient.create()
+
+    val os: PositionedOutputStream = new FSPositionedOutputStream {
+      private[this] val client: AppendBlobClient = appendClient
+
+      override def flush(): Unit = {
+        bb.flip()
+
+        if (bb.limit() > 0) {
+          client.appendBlock(new ByteArrayInputStream(bb.array(), 0, bb.limit()), bb.limit())
+        }
+
+        bb.clear()
+      }
+
+      override def close(): Unit = {
+        if (!closed) {
+          flush()
+          closed = true
+        }
+      }
+    }
+
+    new WrappedPositionedDataOutputStream(os)
+  }
+
+  def delete(filename: String, recursive: Boolean): Unit = {
+    val (account, container, path) = getAccountContainerPath(filename)
+    val blobClient: BlobClient = getBlobClient(account, container, path)
+
+    val fileStatus = this.fileStatus(filename)
+
+    if (recursive) {
+      val blobContainerClient = getContainerClient(account, container)
+
+      val options = new ListBlobsOptions()
+      options.setPrefix(dropTrailingSlash(path) + "/")
+      val prefixMatches = blobContainerClient.listBlobs(options, Duration.ofMinutes(1))
+
+      prefixMatches.forEach(blobItem => {
+        getBlobClient(account, container, blobItem.getName).delete()
+      })
+
+      if (fileStatus.isFile) blobClient.delete()
+    }
+    else {
+        blobClient.delete()
+    }
+  }
+
+  def listStatus(filename: String): Array[FileStatus] = {
+    val (account, container, path) = getAccountContainerPath(filename)
+
+    val blobContainerClient: BlobContainerClient = getContainerClient(account, container)
+    val statList: ArrayBuffer[FileStatus] = ArrayBuffer()
+
+    val prefix = dropTrailingSlash(path) + "/"
+    // collect all children of this directory (blobs and subdirectories)
+    val prefixMatches = blobContainerClient.listBlobsByHierarchy(prefix)
+
+    prefixMatches.forEach(blobItem => {
+      val blobFileName = s"hail-az://$account/$container/${blobItem.getName}"
+      statList += fileStatus(blobFileName)
+    })
+    statList.toArray
+  }
+
+  def glob(filename: String): Array[FileStatus] = {
+    var (account, container, path) = getAccountContainerPath(filename)
+    path = dropTrailingSlash(path)
+
+    globWithPrefix(prefix = s"hail-az://$account/$container", path = path)
+  }
+
+  def fileStatus(account: String, container: String, path: String): FileStatus = {
+    if (path == "") {
+      return new BlobStorageFileStatus(s"hail-az://$account/$container", null, 0, true)
+    }
```

Comment:
This seems bizarre. Why not error if you ask for the blob with no name?

Reply:
This is done because if a user asks for the status of the root blob (which is identified by hail-az://accountName/containerName/) then I don't think you should error

Reply:
I'm with Vedant here but I wouldn't call it the root blob, since it is not a blob. This would in my mind be akin to `ls /`

Reply:
Ah wait no, that would be `path = "/"`. Ya Dan I think this should error. However, this is the current behavior.